### PR TITLE
New Weimer

### LINKED ISSUE #284: plug in the 2005 version Weimer in WAM
After the validation of the neutral temperature during the 2013 and 2015 St. Patrick storms. We found that the Temp. is pretty high (about 3000k at around 250km) at the peak storm time, especially at high latitude and polar region. While the neutral temp. with our new Joule heating factors during quite time are pretty reasonable against MSIS output based on our validation.    We plan to plug-in the new version Weimer (2005) in WAM to bring in the saturation mechanism to control the high peak neutral temp in the current runs.

### PR DESCRIPTION
**Pull request by akubaryk**
As part 1 of #329, a tag has been created for the current tip of development, and we now want to pull new_weimer into development to resolve #284, #296, and #303.

I have a concern about the code that will be noted in a followup comment.

### CODE REVIEW

File: NEMS/src/atmos/phys/efield.f [hunk 1]
```diff
@@ -1089,9 +1100,10 @@ subroutine prep_weimer
        write(iulog,*)  '  tilt =',tilt
       end if
 
-
       call SetModel( angle, bt, tilt, v_sw )
```
Comment (akubaryk) [inline NEMS/src/atmos/phys/efield.f]: Do we still need to call this? Can we remove the old weimer SetModel and EPotVal from efield.f?

File: NEMS/src/atmos/phys/efield.f [hunk 2]
```diff
@@ -366,6 +367,10 @@ subroutine GlobalElPotential
         mlat_90 = 90. - ylatm(ilat)  ! mag. latitude
         do ilon = 0,nmlon
     	  pot  = 1000.*EpotVal( mlat_90, ylonm(ilon)*deg2mlt ) ! calculate potential (kv -> v)
```
Comment (akubaryk) [inline NEMS/src/atmos/phys/efield.f]: See comment below -- is this line necessary? Do we need to keep EpotVal routine in efield.f?

File: NEMS/src/atmos/phys/efield.f (hunk 1 above)
Comment (ZhuxiaoLi66) [reply NEMS/src/atmos/phys/efield.f]: good reminder. do some cleaning up works now.

File: NEMS/src/atmos/phys/efield.f (hunk 2 above)
Comment (ZhuxiaoLi66) [reply NEMS/src/atmos/phys/efield.f]: no

**Comment by ZhuxiaoLi66** [comment]
Adam, I just found that the origin/new_weimer branch on Github can't go though the standalone compiling on Theia. the message indicates that some code 'e.g. idea_solar_init.f' need a more recent F90 compiler. While origin/ipe_refactor_dev_wam can go through.
Have you notice that? It should be a issue or not? would like to make clear about this before push the little bit clean up.

File: NEMS/src/atmos/phys/efield.f (hunk 1 above)
Comment (ZhuxiaoLi66) [reply NEMS/src/atmos/phys/efield.f]: No.

**Comment by ZhuxiaoLi66** [comment]
Since development also can be compiled on Theia, this shouldn't be a big issue after merging, just would like to make sure about it.

**Comment by akubaryk** [comment]
Try running `gmake clean` from NEMS/src and compiling again -- it's probably some leftover pre-compiled stuff from a prior build using a different module set?

**Comment by ZhuxiaoLi66** [comment]
Adam, I pushed the clean up into origin/new_weimer branch. What I should do for next step?

**Comment by akubaryk** [comment]
I think we should still remove SetModel and EPotVal subroutines from efield.f -- I'll push this up and then you can submit a review on the code before the merge can happen.

**Comment by ZhuxiaoLi66** [comment]
I had thought of this yesterday, the code is kind of complicated, we can remove SetModel and EpotVal, and some curtain other subroutines, but not all, and the subroutines has some complicated calling procedures. I had thought to remove the subroutines together when I get time for save reason, while maybe remove SetModel and EpotVal at first is also a good idea.

**Comment by akubaryk** [comment]
I might try ripping out a bunch of it and seeing if we get ~identical results. Would love to pare that code down.

**Comment by ZhuxiaoLi66** [comment]
understand, let us do it.

**Comment by ZhuxiaoLi66** [comment]
I am testing, will return to you soon.

**Comment by akubaryk** [comment]
I'm sure there's bits of additional cleanup that can be done, but that should be the bulk of it. Feel free to submit a review now.

**Comment by ZhuxiaoLi66** [comment]
I reviewed your code changing in Bitwise_identical,  your revise of subroutine of ReadCoef should work.
Other clean up also look fine.   While I don't quite understand your changing in GECMP, especially the first part of it. could you explain it? also it seems not a whole set of that subroutine, since it is a very important subroutine to calculate the electric potential, we need to be very careful to change it except some marked out sentences. Could you give me a whole code of your revised version? thanks.

**Comment by ZhuxiaoLi66** [comment]
Never mind, I pulled your code and having a look again now.

**Comment by ZhuxiaoLi66** [comment]
Great, It seems that you just deleted the marked sentences. Could you also please send me your output? thanks.

**Comment by akubaryk** [comment]
/scratch4/NCEPDEV/stmp3/new_weimer_test_1 and 2 -- 2 is with the base new_weimer branch (before my most recent commit) and 1 is after. I only ran for one hour but bitwise reproducibility is adequate.

**Comment by ZhuxiaoLi66** [comment]
Great work, Adam. I also did a 24 hours SWIN run with the revised new_weimer and have got the bitwise outputs with the former version. 
/scratch3/NCEPDEV/swpc/scrub/Zhuxiao.Li/PTMP/weimer2005_swin_20130101_1day_cleanup_back
/scratch3/NCEPDEV/swpc/scrub/Zhuxiao.Li/PTMP/weimer2005_swin_20130101_1day_cleanup_adam
we took time, while it seems worth it! 
I am going to approve the pull request.

**Pull request merged**